airflow/www/static/js/api/useSetTaskInstanceNotes.ts

```diff
@@ -0,0 +1,107 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import axios from 'axios';
+import { useMutation, useQueryClient } from 'react-query';
+import { getMetaValue, getTask } from 'src/utils';
+import type { GridData } from 'src/api/useGridData';
+import { emptyGridData } from 'src/api/useGridData';
+import type { API } from 'src/types';
+import useErrorToast from '../utils/useErrorToast';
+
+const setTaskInstancesNotesURI = getMetaValue('set_task_instance_note');
+
+export default function useSetTaskInstanceNotes(
+  dagId: string,
+  runId: string,
+  taskId: string,
+  mapIndex: number,
+  newNotesValue: string,
+) {
+  const queryClient = useQueryClient();
+  const errorToast = useErrorToast();
+  // Note: Werkzeug does not like the META URL with an integer. It can not put _MAP_INDEX_ there
```

Comment:
Yes I had some doubts here as well.. 
I felt it's not a good approach to now start splitting up each TaskInstance related endpoint into two. That would lead to code duplication in many places, which would be a shame.
I felt this was the best way of handling it.

Reply:
Just an idea but maybe those 2 entries in the spec can map to the same `x-openapi-router-controller` to not have code duplication.

Reply:
Ohhh nice idea!!! I will give that a try :)!

Reply:
Managed to make it work somewhat. Had to still introduce a different function as it wouldn't be able to show the different URLs otherwise.

Reply:
Hmm. Yeah, if integers don't work in the URL then I guess putting it in the body makes sense.

Reply:
I think either is fine but this is more in line with the getTaskInstance information endpoints. Is it okay to keep is as it is (considering this doesn't need to be set in stone of course :))?